# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/selection/crashtests/selection-modify-line-next-to-input-and-make-it-invisible.html
# Commit: 6d76b71d89f3
# Full Hash: 6d76b71d89f3d3218632b68cecf9c7295b4fb41b
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-04-04 03:44:04
# Regressor Bug: 1887954
# File Overlap Count: 2
# Description:
#   Bug 1887954 - Make `nsIFrame::GetLastLeaf()` never returns a frame in native anonymous subtrees under the given frame r=emilio
#   
#   It does not check whether the child frame is in a native anonymous subtree
#   only when every first child at digging the frame tree.  This patch rewrites
#   the complicated loop with 2 for-loops and make it always check whether the
# ==============================================================================

diff -r 3187c4f491a8 -r 6d76b71d89f3 testing/web-platform/tests/selection/crashtests/selection-modify-line-next-to-input-and-make-it-invisible.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/selection/crashtests/selection-modify-line-next-to-input-and-make-it-invisible.html	Wed Apr 03 07:20:57 2024 +0000
@@ -0,0 +1,23 @@
+<!doctype html>
+<html>
+<head>
+<meta charset="utf-8">
+<script>
+document.addEventListener("DOMContentLoaded", () => {
+  getSelection().addRange(document.createRange());
+  const range = document.createRange();
+  range.selectNode(document.querySelector("meter"));
+  getSelection().addRange(range);
+  getSelection().modify('move', 'left', 'line');
+  document.querySelector("input").style.display = "none";
+  getSelection().modify('move', 'left', 'lineboundary');
+})
+</script>
+</head>
+<body>
+<input>
+<div contenteditable>
+<meter>
+</div>
+</body>
+</html>
